src/event.rs

```diff
@@ -59,6 +59,11 @@ pub enum EventKind {
         database_key: DatabaseKeyIndex,
     },
 
+    WillIterateCycle {
+        /// The database-key for the cycle head. Implements `Debug`.
+        database_key: DatabaseKeyIndex,
```

Comment:
Should we also include the iteration count? I think I'd find this useful to see in logs.

Reply:
Yes, good idea